core/src/cluster_info_vote_listener.rs

```diff
@@ -317,56 +348,101 @@ impl ClusterInfoVoteListener {
+                )
+                .filter_map(|(label, vote, verify_result, packet)| {
+                    if *verify_result != 0 {
+                        Some((vote, (label, packet)))
                     } else {
-                        sigverify::ed25519_verify_cpu(&msgs)
-                    };
-                    assert_eq!(
-                        r.iter()
-                            .map(|packets_results| packets_results.len())
-                            .sum::<usize>(),
-                        votes.len()
-                    );
-                    let valid_votes: Vec<_> = votes
-                        .into_iter()
-                        .zip(r.iter().flatten())
-                        .filter_map(|(vote, verify_result)| {
-                            if *verify_result != 0 {
-                                Some(vote)
-                            } else {
-                                None
-                            }
-                        })
-                        .collect();
-                    vote_txs_sender.send(valid_votes)?;
-                    sigverify::mark_disabled(&mut msgs, &r);
+                        None
+                    }
+                })
+                .unzip();
+
+                verified_vote_transactions_sender.send(vote_txs)?;
+                verified_vote_packets_sender.send(packets)?;
+            }
+
+            sleep(Duration::from_millis(GOSSIP_SLEEP_MILLIS));
+        }
+    }
+
+    fn bank_send_loop(
+        exit: Arc<AtomicBool>,
+        verified_vote_packets_receiver: VerifiedVotePacketsReceiver,
+        poh_recorder: Arc<Mutex<PohRecorder>>,
+        packets_sender: &CrossbeamSender<Vec<Packets>>,
+    ) -> Result<()> {
+        let mut verified_vote_packets = VerifiedVotePackets::default();
+        let mut time_since_lock = Instant::now();
+        let mut update_version = 0;
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                return Ok(());
+            }
+
+            if let Err(e) = verified_vote_packets
+                .get_and_process_vote_packets(&verified_vote_packets_receiver, &mut update_version)
+            {
+                match e {
+                    Error::CrossbeamRecvTimeoutError(RecvTimeoutError::Disconnected) => {
+                        return Ok(());
+                    }
+                    Error::CrossbeamRecvTimeoutError(RecvTimeoutError::Timeout) => (),
+                    _ => {
+                        error!("thread {:?} error {:?}", thread::current().name(), e);
+                    }
+                }
+            }
+
+            if time_since_lock.elapsed().as_millis() > GOSSIP_SLEEP_MILLIS as u128 {
+                let bank = poh_recorder.lock().unwrap().bank();
+                if let Some(bank) = bank {
+                    let last_ts = bank.last_vote_sync.load(Ordering::Relaxed);
+                    let (new_ts, msgs) = verified_vote_packets.get_latest_votes(last_ts);
```

Comment:
This is a version, not a timestamp right?

Reply:
hehe yeah I changed it in one of these earlier commits and missed this spot, thanks!